RedHatInsights/ocp-advisor-frontend
# feat(OCPADVISOR-70): Enable "Upgrade Risks" tab on the cluster details view

Implements https://issues.redhat.com/browse/OCPADVISOR-70.
Sketched at https://www.sketch.com/s/c5c07e06-f4a1-4c64-9cdf-8b12d8ac9ad5/a/8ylYyDw.

This makes the recommendations table render in a tab on the cluster details view (/clusters/%id) and adds the second tab called "Upgrade Risks." The latter temporarily uses placeholder content.

"Upgrade Risks" tab always rendered in beta (preview) release. The tab is displayed in stable if the feature flag `ocp-advisor.upgrade-risks.enable-in-stable` is set to true.

**I will need to refactor component tests for the Cluster component a bit so I decided not to include tests in this PR.**

## How to test:

1. Navigate to any cluster and make sure the main content is rendered in tabs.
2. Tabs support URL search parameter "active_tab": you can play with it and make sure it doesn't break the page (by default, recommendations are shown unless "active_tab=upgrade_risks" is given.
3. You can build the PR under stable release (stage-stable or prod-stable) and try to toggle the Unleash feature flag. When disabled, the tab must not be shown.

## Screenshots

![image](https://user-images.githubusercontent.com/31385370/230324000-040a7c4e-db9e-4c34-ae39-9fe75ded8a50.png)
